common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
```

Comment:
This interface changed from something metadata-store specific, into a global thing used to get all kinds of secrets. I think that's cool, but, how is the user supposed to set the global PasswordProvider they want to use? I see some Guice code binding it based on the value of `druid.metadata.storage.connector.password` but this seems kind of weird. Why would the global PasswordProvider be set through the metadata store properties? The ability to do so should be preserved for compatibility but there should also be a "nice" way to do it. Maybe `druid.secret.provider.type=whatever`.

Maybe consider this design instead.

- Add a new thing, SecretProvider, that is a global object for getting secrets.
- Make a SecretProviderPasswordProvider for the metadata store that causes the metadata connector to use the SecretProvider. Make this the default if no password is set for the metadata store.
- If someone sets `druid.metadata.storage.connector.password` then it would still create whatever kind of PasswordProvider they want, and that interface wouldn't change (but is really just there for legacy compatibility).